### PR TITLE
vcmi: remove MAXED_SPELL bonus

### PR DESCRIPTION
**Pull request by rilian-la-te**
It is covered by SPECIAL_FIXED_VALUE_ENCHANT, so, just replace remaining instances and remove bonus altogether.

### CODE REVIEW

**Comment by IvanSavenko** [comment]
No objections against change itself, but perhaps you should also remove enum value from HeroBonus.h?
(although removing enum value will break saves - probably better to rename it to UNUSED_BONUS or something)

Quick search also shows a few more hits for this bonus name than you have in your changes.

**Comment by rilian-la-te** [comment]
@IvanSavenko removed remaining usage. Renamed enum value to UNUSED0.

**Comment by DjWarmonger** [comment]
We can remove it from Bonus enum, but this will break save compatibility. This also requires bump of SERIALIZATION_VERSION.  So this merge can wait for some convenient moment.

**Comment by rilian-la-te** [comment]
@DjWarmonger Ok, bonus is removed. What do you think about splitting this PR, and where JSON part will be merged right away, save compatibility will wait?

**Comment by DjWarmonger** [comment]
> @DjWarmonger Ok, bonus is removed. What do you think about splitting this PR, and where JSON part will be merged right away, save compatibility will wait?
This is actually a good idea, we can do the cleaning sometime later.

**Review by Nordsoft91 — APPROVED** [review]

File: lib/HeroBonus.h [hunk 1]
```diff
@@ -285,7 +285,6 @@ class DLL_LINKAGE CAddInfo : public std::vector<si32>
 	BONUS_NAME(SPELL_DAMAGE) /*val = value*/\
 	BONUS_NAME(SPECIFIC_SPELL_DAMAGE) /*subtype = id of spell, val = value*/\
 	BONUS_NAME(SPECIAL_BLESS_DAMAGE) /*val = spell (bless), additionalInfo = value per level in percent*/\
-	BONUS_NAME(MAXED_SPELL) /*val = id. deprecated in favour of SPECIAL_FIXED_VALUE_ENCHANT*/\
```
Comment (nullkiller) [inline lib/HeroBonus.h:288]: Interesting if we break save compatibility here as bonus enum numeration may shift?

Comment (nullkiller) [reply lib/HeroBonus.h:288]: If yes I would rename bonus instead as DEPRECATED_MAXED_SPELL or UNUSED_MAXED_SPELL and remove when we break save compatibility once again. (Or do we plan to do this in 1.2?). Or at least increase serialization version.

**Comment by nullkiller** [comment]
So this waits when we decide if we want to break save compatibility in 1.2?

**Comment by rilian-la-te** [comment]
@nullkiller this bonus is already deprecated. If we break save compatibility, I think we can rebase it and remove bonus altogether.

**Comment by rilian-la-te** [comment]
Handled in #1518

**Pull request closed without merging**